### PR TITLE
Add error explanation for E0708

### PR DESCRIPTION
**Pull request by bishtpawan**
Add long explanation for the E0708 error code
Part of #61137

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0708.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+async` non-`move` closures with parameters are not currently supported
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0708.md]: I think a backtick is missing at the beginning of the sentence.

Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0708.md]: Also missing a dot at the end of the sentence.

File: src/librustc_error_codes/error_codes/E0708.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Erroneous code example:
+
+```compile_fail,E0710
+#![feature(async_closure)]
+
+fn main() {
+    let add_one = async |num: u8| {
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0708.md]: You need to add a comment (at minimum `// error!`) on the failing line to help users identify the problem more visually.

File: src/librustc_error_codes/error_codes/E0708.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+`async` with non-move currently not supported with the current
+version, you can use successfully by using move:
+
+```
+#![feature(async_closure)]
+
+fn main() {
+    let add_one = async move |num: u8| {
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0708.md]: And just like you have to add `// error!`, you need to add a comment like `// ok!` on the fixed lines.

**Comment by bishtpawan** [comment]
got all the points. as now I'm closing this PR and will raise new one as I found some changes in it. Thanks

**Pull request closed without merging**